### PR TITLE
fix(girderclient): file/chunk endpoint changes for girder 3

### PR DESCRIPTION
**Pull request by aronhelser**
No longer uses multi-part requests.

CC @psavery

### CODE REVIEW

**Comment by aronhelser** [comment]
Would be good, after this is merged, to tag a v1.0.1 version with the two Girder 3 fixes.

File: cumulus/girderclient.py [hunk 1]
```diff
@@ -117,18 +117,15 @@ def _upload_file(self, name, path, parent_id):
...
-                    fields=[('uploadId',  upload_id),
-                            ('offset', str(uploaded)),
-                            ('chunk', (name, part, 'application/octet-stream'))]
+                params['uploadId'] = upload_id
+                params['offset'] = uploaded
 
-                )
 
                 headers = self._headers.copy()
                 headers['Content-Type'] = m.content_type
```
Comment (patrickoleary) [inline cumulus/girderclient.py]: headers['Content-Type'] = m.content_type 

is wrong. You deleted m for the MultipartEncoder. It should be

headers['Content-Type'] = 'application/octet-stream'

File: cumulus/girderclient.py [hunk 2]
```diff
@@ -117,18 +117,15 @@ def _upload_file(self, name, path, parent_id):
 
                 part = fp.read(chunk_size)
 
-                m = MultipartEncoder(
```
Comment (patrickoleary) [inline cumulus/girderclient.py]: you should remove line 22

from requests_toolbelt import MultipartEncoder

since you've removed the need for it here.

Comment (patrickoleary) [reply cumulus/girderclient.py]: also requests-toolbelt was only used here in the entire repository. So probably should remove it from the requirements.txt file.

File: cumulus/girderclient.py (hunk 1 above)
Comment (patrickoleary) [reply cumulus/girderclient.py]: This should fix the failing test.

File: cumulus/girderclient.py [hunk 3]
```diff
@@ -117,18 +117,15 @@ def _upload_file(self, name, path, parent_id):
...
-                    fields=[('uploadId',  upload_id),
-                            ('offset', str(uploaded)),
-                            ('chunk', (name, part, 'application/octet-stream'))]
+                params['uploadId'] = upload_id
+                params['offset'] = uploaded
 
-                )
```
Comment (patrickoleary) [inline cumulus/girderclient.py]: remove one of the two blank lines, at least.

**Comment by patrickoleary** [comment]
@cjh1

**Comment by cjh1** [comment]
Can we fix up the CI? ( I know it doesn't look related to the these changes  )

**Comment by aronhelser** [comment]
Just looking at CI, getting this failure I'm not sure how to address:

> StateTransitionBehavior
    ✓ does nothing if there's no taskflow
08 07 2019 16:37:45.773:WARN [web-server]: 404: /api/v1/simulations/574c8aa00640fd3f1a3b379f
    fs actions depending on taskflow state
      ✗ does not update folders if not all complete
	Error: spy was not supposed to be called
	    at assert (test/contexts/tests.all.js:8130:9)
	    at Expectation.toNotHaveBeenCalled (test/contexts/tests.all.js:7737:28)
	    at UserContext.<anonymous> (test/contexts/tests.all.js:125576:52)
	    at <Jasmine>

**Comment by cjh1** [comment]
Looks like the flake8-doctrings deps are currently [broken](https://gitlab.com/pycqa/flake8-docstrings/issues/36)

**Comment by cjh1** [comment]
I think pinning to `pydocstyle<4` should solve this issue.

**Comment by aronhelser** [comment]
@cjh1 There we go - ok to merge? Bot wants a review....

**Pull request merged**